### PR TITLE
Add other inputs for redis connection

### LINKED ISSUE #255: fakeredis does not support all the redis connection arguments
Some unsupported arguments include socket_timeout, socket_connect_timeout, socket_keepalive, socket_keepalive_options, socket_type, retry_on_timeout, socket_read_size and health_check_interval. While I wouldn't expect this to necessarily be simulated by fakeredis it would be nice to be able to use this in the from_url command.

Ex: FakeRedis.from_url(redis_url, db=None, socket_connect_timeout=5, socket_timeout=30)

### PR DESCRIPTION
**Pull request by nateyoder**
Closes #255

### CODE REVIEW

File: fakeredis/_server.py [hunk 1]
```diff
@@ -2490,12 +2490,22 @@ class FakeConnection(redis.Connection):
...
+                 socket_timeout=None, socket_connect_timeout=None,
+                 socket_keepalive=False, socket_keepalive_options=None,
+                 socket_type=0, retry_on_timeout=False,
                  encoding='utf-8', encoding_errors='strict',
                  decode_responses=False,
                  health_check_interval=0):
         self.pid = os.getpid()
         self.db = db
         self.password = password
+        # Allow socket attributes to be passed in and saved  even if they aren't used
```
Comment (bmerry) [inline fakeredis/_server.py]:
```suggestion
        # Allow socket attributes to be passed in and saved even if they aren't used
```

File: fakeredis/_server.py [hunk 2]
```diff
@@ -2490,12 +2490,22 @@ class FakeConnection(redis.Connection):
     description_format = "FakeConnection<db=%(db)s>"
 
     def __init__(self, server, db=0, password=None,
+                 socket_timeout=None, socket_connect_timeout=None,
+                 socket_keepalive=False, socket_keepalive_options=None,
+                 socket_type=0, retry_on_timeout=False,
```
Comment (bmerry) [inline fakeredis/_server.py]: Are these in the same position/order that the redis `__init__` takes them?

File: fakeredis/_server.py [hunk 3]
```diff
@@ -2490,12 +2490,22 @@ class FakeConnection(redis.Connection):
...
                  decode_responses=False,
                  health_check_interval=0):
         self.pid = os.getpid()
         self.db = db
         self.password = password
+        # Allow socket attributes to be passed in and saved  even if they aren't used
+        self.socket_timeout = socket_timeout
+        self.socket_connect_timeout = socket_connect_timeout or socket_timeout
+        self.socket_keepalive = socket_keepalive
+        self.socket_keepalive_options = socket_keepalive_options or {}
```
Comment (bmerry) [inline fakeredis/_server.py]: I take it this is equivalent to what real redis is setting on `self`?

File: fakeredis/_server.py (hunk 2 above)
Comment (nateyoder) [reply fakeredis/_server.py]: Yeah [at least as it is currently on master](https://github.com/andymccurdy/redis-py/blob/656b551eb247c491976646092a71320407c0fa03/redis/connection.py#L488).  I had forgotten a couple and just added them. Sorry.

File: fakeredis/_server.py (hunk 3 above)
Comment (nateyoder) [reply fakeredis/_server.py]: Yeah this was taken directly from [redis-py](https://github.com/andymccurdy/redis-py/blob/656b551eb247c491976646092a71320407c0fa03/redis/connection.py#L503).

**Comment by bmerry** [comment]
Thanks, I'm going to merge. Don't worry about coveralls - it was also claiming a reduction in coverage on another PR that only touched the README, so I don't believe in it.

**Pull request merged**